### ISSUE TITLE
-- Hangup --

### ISSUE THREAD

**Issue by ghost** [orakaro/rainbowstream#199]
Could be due to this

http://www.theregister.co.uk/2016/10/21/dns_dyn_ddos/
Dyn dinged by DDoS: US DNS firm gives web a bad hair day
Reddit, Github, Airbnb and pals affected

**Comment by stephenwithav** [comment]
Are you still getting the Hangup error?

I just installed rainbowstream and the hangup error made me think it wasn't working.

**Comment by manasb** [comment]
@stephenwithav  Yeah, I have been seeing that. I have not tested in a few hours, let me check and get back to you.

**Comment by stephenwithav** [comment]
Thank you for the update.  I'm receiving hangups fairly quickly, frequently within a minute before anything is streamed.

**Comment by manasb** [comment]
I'm on Debian Testing, by the way.

```
$  pip3 show rainbowstream
---
Metadata-Version: 2.0
Name: rainbowstream
Version: 1.3.6
Summary: A smart and nice Twitter client on terminal.
Home-page: http://www.rainbowstream.org/
Author: Vu Nhat Minh
Author-email: nhatminh179@gmail.com
Installer: pip
License: MIT License
Location: /home/manas/.local/lib/python3.5/site-packages
Requires: twitter, PySocks, pyfiglet, arrow, Pillow, requests, python-dateutil, pocket
Classifiers:
  Development Status :: 5 - Production/Stable
  Environment :: Console
  Intended Audience :: End Users/Desktop
  Natural Language :: English
  Operating System :: OS Independent
  Programming Language :: Python :: 2.7
  Programming Language :: Python :: 3.2
  Programming Language :: Python :: 3.3
  Programming Language :: Python :: 3.4
  Programming Language :: Python :: 3.5
  Topic :: Internet :: WWW/HTTP :: Dynamic Content :: CGI Tools/Libraries
  Topic :: Utilities
  License :: OSI Approved :: MIT License
Entry-points:
  [console_scripts]
  rainbowstream=rainbowstream.rainbow:fly

```

**Comment by violetina** [comment]
i can confirm this too, even with -to 300 (timeout 300sec) it times out after few seconds, it did work fine before tho

**Comment by manasb** [comment]
@violetina @DTVD  I've tried making a new key multiple times. I am still receiving the Hangup message.

**Comment by DTVD** [comment]
My client is working without problem. Twitter API status seems backed to normal: https://dev.twitter.com/overview/status

Are you guy still having hangup error ?

**Comment by manasb** [comment]
@DTVD Hi,

Yes, I am still seeing the Hangup error.

**Comment by stephenwithav** [comment]
@DTVD still experiencing here, too.

Would you object to a PR that reconnects on Hangup?

**Comment by maikell** [comment]
I can confirm this bug on Centos 7, Fedora 24 and Debian Jessie. "Hangup" message after Friday.

**Comment by stephenwithav** [comment]
I re-authorized the app, initially received a hangup, but now it's working for me.

**Comment by maikell** [comment]
I can confirm. Nice work!

**Comment by maikell** [comment]
It worked for about 15 minutes :)
"There seems to be a connection problem.
You might want to check your proxy settings (host, port and type)!"

**Comment by stephenwithav** [comment]
Mine is still going strong.  I'm afraid to touch it. :fearful:

**Comment by stephenwithav** [comment]
Switched themes, now Hangups again.

**Comment by manasb** [comment]
@stephenwithav @maikell I've messaged twitter.com/@Support about this. I didn't hear anything back. We might want to ping them again.

**Comment by manasb** [comment]
Hi I am in https://gitter.im/DTVD/rainbowstream if anyone would like to co-ordinate figuring out this error

:)

**Comment by DTVD** [comment]
Thanks @manasb @stephenwithav and @maikell for reporting. As far as I know this is result of [TCP connection break](https://github.com/sixohsix/twitter/blob/master/twitter/stream.py#L179) with Twitter Stream API. Currently I can not reproduce this problem but if possible can you guy try and give result of following steps ?

* A screenshot when Hangup
* Output of `g["traceback"]` when Hangup. You can type this directly into RS's console.
* Is other client works well with stream ?
* Try to _revoke access_ of RS from your account's [Twitter setting page]
(https://twitter.com/settings/applications) and start RS again.

**Comment by manasb** [comment]
@DTVD 

Hi. Yesterday, I tried revoking & re-authorizing rainbowstream. That did not work.

Seems to be functional for now. I will update if necessary.

**Comment by maikell** [comment]
Is it an idea to change the application name and re-authorize the application? This is so we can check if Twitter actively blocks this CLI application. Will dig around in the code today to try out.

**Comment by DTVD** [comment]
You can fork and clone a version of RS yourself. Just need to grab new Application’s API key and secret. See "Development" section of Readme

**Comment by maikell** [comment]
Ok, this is strange. It seems to work again in Debian Jessie, however it failed on me this morning in Fedora 24. I'll reinstall it tonight and report back if it works.
